waspc/src/Wasp/Analyzer/AnalyzeError.hs

```diff
@@ -18,12 +20,8 @@ data AnalyzeError
   | EvaluationError EE.EvaluationError
   deriving (Show, Eq)
 
-getErrorMessage :: AnalyzeError -> String
-getErrorMessage (ParseError e) = "Parse error:\n" ++ indent 2 (PE.getErrorMessage e)
-getErrorMessage (TypeError e) = "Type error:\n" ++ error "TODO"
-getErrorMessage (EvaluationError e) = "Evaluation error:\n" ++ error "TODO"
-
-getErrorSourcePosition :: AnalyzeError -> SourcePosition
-getErrorSourcePosition (ParseError e) = PE.getSourcePosition e
-getErrorSourcePosition (TypeError e) = error "TODO"
-getErrorSourcePosition (EvaluationError e) = error "TODO"
+getErrorMessageAndCtx :: AnalyzeError -> (String, Ctx)
```

Comment:
Ha, good question :D! Truth is I haven't thought of it, but looking at it now I actually feel like it might be ok to leave it as it is and not use WithCtx.

Explanation:
This is "external" facing interface, "external" meaning that other parts of code using Analyzer will be calling this function, `getErrorMessageAndCtx`. `WithCtx` on the other hand is an internal thing of Analyzer, internal in the sense that so far it is used only in the Analyzer, and while I could expose it here, out to the world, it wouldn't bring much value, since not so much will be done with output of `getErrorMessageAndCtx`, not much passing around and decomposing. So, introducing new concept to the outside vs value it brings -> I think it might be ok to not use WithCtx here.

Reply:
Makes sense, thanks for the explanation!